master/pkg/model/task.go

```diff
@@ -88,14 +88,18 @@ func (a AllocationID) String() string {
 type Allocation struct {
 	bun.BaseModel `bun:"table:allocations"`
 
-	AllocationID AllocationID     `db:"allocation_id" bun:"allocation_id,pk"`
-	TaskID       TaskID           `db:"task_id" bun:"task_id,notnull"`
-	Slots        int              `db:"slots" bun:"slots,notnull"`
-	ResourcePool string           `db:"resource_pool" bun:"resource_pool,notnull"`
-	StartTime    *time.Time       `db:"start_time" bun:"start_time"`
-	EndTime      *time.Time       `db:"end_time" bun:"end_time"`
-	State        *AllocationState `db:"state" bun:"state"`
-	IsReady      *bool            `db:"is_ready" bun:"is_ready"`
+	AllocationID               AllocationID     `db:"allocation_id" bun:"allocation_id,pk"`
+	TaskID                     TaskID           `db:"task_id" bun:"task_id,notnull"`
+	Slots                      int              `db:"slots" bun:"slots,notnull"`
+	ResourcePool               string           `db:"resource_pool" bun:"resource_pool,notnull"`
+	StartTime                  *time.Time       `db:"start_time" bun:"start_time"`
+	EndTime                    *time.Time       `db:"end_time" bun:"end_time"`
+	State                      *AllocationState `db:"state" bun:"state"`
+	IsReady                    *bool            `db:"is_ready" bun:"is_ready"`
+	DTrainPort                 int              `db:"dtrain_port" bun:"dtrain_port,notnull"`
```

Comment:
here and everywhere, i think we should convert these ports from struct fields to a single map[name]port